src/Phinx/Console/Command/Init.php

```diff
@@ -53,55 +57,79 @@ protected function configure()
     /**
      * Initializes the application.
      *
-     * @param \Symfony\Component\Console\Input\InputInterface $input
-     * @param \Symfony\Component\Console\Output\OutputInterface $output
-     * @throws \RuntimeException
-     * @throws \InvalidArgumentException
+     * @param InputInterface  $input  Interface implemented by all input classes.
+     * @param OutputInterface $output Interface implemented by all output classes.
+     *
+     * @throws RuntimeException
+     * @throws InvalidArgumentException
      * @return void
      */
     protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $path = $this->resolvePath($input);
+        $this->writeConfig($path);
+
+        $output->writeln("<info>created</info> {$path}");
+    }
+
+    /**
+     * Return valid $path for Phing's config file.
+     *
+     * @param InputInterface $input Interface implemented by all input classes.
+     *
+     * @return string
+     */
+    protected function resolvePath(InputInterface $input)
     {
         // get the migration path from the config
         $path = $input->getArgument('path');
 
-        if ($path === null) {
-            $path = getcwd();
+        // Fallback
+        if (empty($path)) {
+            $path = getcwd() . DIRECTORY_SEPARATOR . self::FILE_NAME;
         }
 
-        $path = realpath($path);
-
-        if (!is_writable($path)) {
-            throw new \InvalidArgumentException(sprintf(
-                'The directory "%s" is not writable',
-                $path
-            ));
+        // Adding file name if necessary
+        if (is_dir($path)) {
+            $path .= DIRECTORY_SEPARATOR . self::FILE_NAME;
         }
 
-        // Compute the file path
-        $fileName = 'phinx.yml'; // TODO - maybe in the future we allow custom config names.
-        $filePath = $path . DIRECTORY_SEPARATOR . $fileName;
+        // Check if path is available
+        $dirname = dirname($path);
+        if (is_dir($dirname) && !is_file($path)) {
+            return $path;
+        }
 
-        if (file_exists($filePath)) {
-            throw new \InvalidArgumentException(sprintf(
-                'The file "%s" already exists',
-                $filePath
-            ));
+        // Path is valid, but file already exists
+        if (is_file($path)) {
+            throw new InvalidArgumentException('Config file already exists.');
         }
 
+        // Dir is invalid
+        throw new InvalidArgumentException('Invalid path for config file.');
```

Comment:
You could tell people the actual directory names.

Reply:
Right, I will add the file's path.